### PR TITLE
PHOENIX-5879 Fix recently introduced python 2.7 incompatibilities and…

### PR DESCRIPTION
**Pull request by stoty**
… flake8 warning

* downgrade dependencies to make sure setup works on 2.7
* fix python27 incompatibilities
* fix flake8 warnings

additional improvements:
* pare down dependencies
* dockerize tox tests
* update docs
* expose ways to start a local test PQS server from maven

### CODE REVIEW

**Comment by joshelser** [comment]
Nice, this looks good. I'll try to run through it myself.

File: python/phoenixdb/README.rst [hunk 1]
```diff
@@ -65,17 +56,11 @@ necessary requirements::
...
-quickly using `Docker <https://www.docker.com/>`_::
+You can start a Phoenix QueryServer test instance on http://localhost:8765 for testing by running
+the following command in the phoenix-queryserver directory:
 
-    docker-compose up
-
-Or if you need an older version of Phoenix::
-
-    PHOENIX_VERSION=4.9 docker-compose up
+    mvn clean verify -am -pl queryserver-it -Dtest=foo -Dit.test=QueryServerBasicsIT#startLocalPQS \
```
[the remark this answers is not in the document]
Comment (stoty) [reply python/phoenixdb/README.rst:62]: Done

File: python/phoenixdb/README.rst [hunk 2]
```diff
@@ -85,12 +70,15 @@ the `PYTHONPATH` environment variable to point to the library directly::
...
+Don't forget to run flake8 on your changes.
+
 Interactive SQL shell
 ---------------------
 
 There is a Python-based interactive shell include in the examples folder, which can be
 used to connect to Phoenix and execute queries::
 
+    pip install -r ./examples/requirements.txt
     ./examples/shell.py http://localhost:8765/
```
[the remark this answers is not in the document]
Comment (stoty) [reply python/phoenixdb/README.rst:82]: No, I didn't check it.

I've deleted it, I don't think it's very relevant.

**Comment by stoty** [comment]
I've updated the author info, but I'd be happy to change it, if you can suggest a better author string.

File: python/phoenixdb/README.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshelser) [reply python/phoenixdb/README.rst]: Sounds good.

File: python/phoenixdb/README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshelser) [reply python/phoenixdb/README.rst]: Thank you :)

**Comment by joshelser** [comment]
> I've updated the author info, but I'd be happy to change it, if you can suggest a better author string.

Thanks! I think "Apache Software Foundation" is just fine. The other idea would be to say "Apache Phoenix", but they're more or less the same thing.

This is fantastic +1

**Comment by stoty** [comment]
Thanks for thorough review @joshelser

**Pull request closed without merging**